vhdbuilder/packer/cleanup.sh

```diff
@@ -193,4 +245,22 @@ if [[ -n "${AZURE_RESOURCE_GROUP_NAME}" && "${DRY_RUN}" == "False" ]]; then
           echo "Deletion completed"
       fi
   done
-fi
\ No newline at end of file
+fi
+
+# clean up old packer and vnet resource groups created over a week ago that didn't get cleaned in the building vhd step in case the pipeline fails
+if [[ -n "${AZURE_RESOURCE_GROUP_NAME}" && "${DRY_RUN}" == "True" ]]; then
+  pkr_groups=$(az group list | jq --arg dl $deadline '.[] | select(.name | test("pkr-Resource-Group*")) | select(.tags.now < $dl).name' | tr -d '\"' || "")
+  for pkr_group in $pkr_groups; do
+      echo "Deleting packer resource group $pkr_group"
+      az group delete --name ${pkr_group} --yes 
+      echo "Deleted packer resource group $pkr_group"
+  done
+
+  # the old vnet groups do not have "now" in tags
```

Comment:
maybe only do this in the windows case since we don't have these RGs for linux builds?